legal-api/src/legal_api/core/meta/filing.py

```diff
@@ -502,14 +502,17 @@ def alter_outputs_alteration(filing, outputs):
     def alter_outputs_correction(filing, business, outputs):
         """Handle output file list modification for corrections."""
         if filing.filing_type == 'correction':
+            if filing.meta_data.get('correction', {}).get('toLegalName'):
+                outputs.add('certificateOfNameChange')
             corrected_filing_id = filing.filing_json.get('correction', {}).get('correctedFilingId')
             original_filing = FilingStorage.find_by_id(corrected_filing_id)
             if is_special_resolution_correction(
                     filing.filing_json['filing'], business, original_filing
                     ):
                 if filing.filing_json['filing']['correction'].get('rulesFileKey'):
                     outputs.add('certifiedRules')
-                outputs.add('specialResolution')
+                if filing.meta_data.get('correction', {}).get('hasResolutionChanges'):
+                    outputs.add('specialResolution')
```

Comment:
Hiding this if there are no resolution changes. Only documents that show change hopefully.